consumer/migration/hermes_migrator.go

```diff
@@ -0,0 +1,229 @@
+	if err != nil {
+		return fmt.Errorf("channel status error: %w", err)
+	}
+
+	log.Debug().Msgf("Channel status: %s", statusResponse.Status)
+	if statusResponse.Status == registry.ChannelStatusNotFound {
+		if err := m.transactor.OpenChannel(chainID, id, activeHermes.Hex(), registryAddress.Hex()); err != nil {
+			return fmt.Errorf("open new channel error: %w", err)
+		}
+		err = m.waitForChannelOpened(chainID, common.HexToAddress(id), activeHermes, registryAddress, openChannelTimeout)
+	} else if statusResponse.Status == registry.ChannelStatusInProgress {
+		err = m.waitForChannelOpened(chainID, common.HexToAddress(id), activeHermes, registryAddress, openChannelTimeout)
+	}
+	if err != nil {
+		return fmt.Errorf("error during waiting for channel opening: %w", err)
+	}
+
+	channelImpl, err := m.addressProvider.GetActiveChannelImplementation(chainID)
+	if err != nil {
+		return fmt.Errorf("error during getting channel implementation: %w", err)
+	}
+	newChannel, err := crypto.GenerateChannelAddress(id, activeHermes.Hex(), registryAddress.Hex(), channelImpl.Hex())
+	if err != nil {
+		return fmt.Errorf("generate channel address erro: %w", err)
+	}
+
+	log.Debug().Msgf("Send transaction. Old Hermes: %s, new Hermes %s (channel: %s)", oldHermes, activeHermes, newChannel)
+
+	return m.hps.Withdraw(chainID, chainID, identity.FromAddress(id), oldHermes, common.HexToAddress(newChannel), nil)
+}
+
+// IsMigrationRequired check whether migration required
+func (m *HermesMigrator) IsMigrationRequired(id string) (bool, error) {
+	chainID := config.GetInt64(config.FlagChainID)
+	activeHermes, err := m.addressProvider.GetActiveHermes(chainID)
+	if err != nil {
+		return false, fmt.Errorf("could not get hermes address: %w", err)
+	}
+	knownHermeses, err := m.addressProvider.GetKnownHermeses(chainID)
+	if err != nil {
+		return false, fmt.Errorf("could not get hermes address: %w", err)
+	}
+	oldHermeses := getOldHermeses(knownHermeses, activeHermes)
+	if len(oldHermeses) != 1 {
+		return false, nil
+	}
+	oldHermes := oldHermeses[0]
+
+	registryAddress, err := m.addressProvider.GetRegistryAddress(chainID)
+	if err != nil {
+		return false, fmt.Errorf("could not get registry address: %w", err)
+	}
+
+	statusResponse, err := m.transactor.ChannelStatus(chainID, id, activeHermes.Hex(), registryAddress.Hex())
+	if err != nil {
+		return false, err
+	}
+	if statusResponse.Status == registry.ChannelStatusNotFound || statusResponse.Status == registry.ChannelStatusInProgress {
+		return true, nil
+	}
+
+	oldBalance, err := m.getBalance(chainID, oldHermes.Hex(), id)
+	if err != nil {
+		return false, fmt.Errorf("error during getting balance: %w", err)
+	}
+	newBalance, err := m.getBalance(chainID, activeHermes.Hex(), id)
+	if err != nil {
+		return false, fmt.Errorf("error during getting balance: %w", err)
+	}
+
+	return crypto.FloatToBigMyst(oldBalanceMigrationMinimumMyst).Cmp(oldBalance) < 0 && newBalance.Cmp(new(big.Int)) == 0, nil
```

Comment:
I think it's in the notes, but do we not want to migrate if new channel has money but old also has?